drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2023 Jan Henke <Jan.Henke@taujhe.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sparkfun_serlcd
+
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/auxdisplay.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(auxdisplay_serlcd, CONFIG_AUXDISPLAY_LOG_LEVEL);
+
+/*
+ * | in ASCII, used to begin a display command
+ */
+#define SERLCD_BEGIN_COMMAND 0x7C
+
+/*
+ * special command for the underlying display controller
+ */
+#define SERLCD_BEGIN_SPECIAL_COMMAND 0xFE
+
+union auxdisplay_display_state {
+	uint8_t byte;
+	struct __packed {
+#ifdef CONFIG_LITTLE_ENDIAN
+		uint8_t padding: 5;
+		bool power: 1;
+		bool cursor: 1;
+		bool blinking: 1;
+#else
+		bool blinking: 1;
+		bool cursor: 1;
+		bool power: 1;
+		uint8_t padding: 5;
+#endif
+	};
+};
+
+struct auxdisplay_serlcd_data {
+	union auxdisplay_display_state display_state;
+	uint16_t cursor_x;
+	uint16_t cursor_y;
+};
+
+struct auxdisplay_serlcd_config {
+	struct auxdisplay_capabilities capabilities;
+	struct i2c_dt_spec bus;
+};
+
+enum auxdisplay_serlcd_command {
+	SERLCD_COMMAND_SET_CUSTOM_CHAR = 0x1B,
+	SERLCD_COMMAND_WRITE_CUSTOM_CHAR = 0x23,
+	SERLCD_COMMAND_CLEAR = 0x2D,
+};
+
+enum auxdisplay_serlcd_special_command {
+	SERLCD_SPECIAL_RETURN_HOME = 0x02,
+	SERLCD_SPECIAL_DISPLAY_CONTROL = 0x08,
+	SERLCD_SPECIAL_SET_DD_RAM_ADDRESS = 0x80,
+};
+
+static int auxdisplay_serlcd_send_command(const struct device *dev,
+					  const enum auxdisplay_serlcd_command command)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+	const uint8_t buffer[2] = {SERLCD_BEGIN_COMMAND, command};
+
+	int rc = i2c_write_dt(&config->bus, buffer, sizeof(buffer));
+
+	/*
+	 * commands need some time to process
+	 */
+	k_sleep(K_MSEC(10));
```

Comment:
add a define for `10`, give it a name like `SERLCD_COMMAND_DELAY`, then your comment become redundant as this would self-describe and you won't need it anymore

Reply:
Done